lib/matplotlib/image.py

```diff
@@ -56,11 +56,67 @@
     'blackman': _image.BLACKMAN,
 }
 
-
 interpolations_names = set(six.iterkeys(_interpd_))
 
 
-class _AxesImageBase(martist.Artist, cm.ScalarMappable):
+def composite_images(images, renderer, magnification=1.0):
+    """
+    Composite a number of RGBA images into one.
+
+    Parameters
+    ----------
+    images : list of objects with `make_image` method
+
+    renderer : RendererBase instance
+
+    magnification : float
+        The additional magnification to apply for the renderer in use.
+    """
+    if len(images) == 0:
+        return np.empty((0, 0, 4), dtype=np.uint8), 0, 0
+
+    parts = []
+    bboxes = []
+    for image in images:
+        data, x, y, trans = image.make_image(renderer, magnification)
+        if data is not None:
+            x *= magnification
+            y *= magnification
+            parts.append((data, x, y, image.get_alpha() or 1.0))
+            bboxes.append(
+                Bbox([[x, y], [x + data.shape[1], y + data.shape[0]]]))
+
+    if len(parts) == 0:
+        return None, 0, 0
+
+    bbox = Bbox.union(bboxes)
+
+    output = np.zeros(
+        (int(bbox.height), int(bbox.width), 4), dtype=np.uint8)
+
+    for data, x, y, alpha in parts:
+        trans = Affine2D().translate(x - bbox.x0, y - bbox.y0)
+        _image.resample(data, output, trans, _image.NEAREST,
+                        resample=False, alpha=alpha)
+
+    return output, bbox.x0 / magnification, bbox.y0 / magnification
+
+
+def _rgb_to_rgba(A):
+    """
+    Convert an RGB image to RGBA, as required by the image resample C++
+    extension.
+    """
+    rgba = np.zeros((A.shape[0], A.shape[1], 4), dtype=A.dtype)
+    rgba[:, :, :3] = A
+    if rgba.dtype == np.uint8:
```

Comment:
`dtype.kind` isn't exactly the right thing.  We could be more generic by handling the int case with `np.iinfo(dtype).max` instead of hardcoding 255 (and thus only supporting 8-bit).  In practice, it's moot, though, since we only support 8-bit RGBA on the C++ extension side anyway (and this assumption is scattered all over anyway).